src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -177,8 +178,7 @@ protected void handlePut(final RestChannel channel, final RestRequest request, f
 
 		final String name = request.param("name");
 
-		if (name == null || name.length() == 0) {
-			badRequestResponse(channel, "No " + getResourceName() + " specified.");
+		if (isResourceNameInvalid(channel, name)) {
```

Comment:
what happens to existing users that are using special characters in their name?

Reply:
Users can still GET and DELETE resources. The restriction is on PUT. They will have to DELETE and PUT resource again with validated names if they want to modify the same resource 

There are characters which are blocked in the UI already https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/master/public/apps/configuration/sections/internalusers/controller.js#L118 . Same restriction is not applicable in the REST API

Reply:
It is UI that should follow REST API restrictions, not reverse.